packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRoot.ts

```diff
@@ -39,3 +39,48 @@ export async function beaconBlocksMaybeBlobsByRoot(
   // and here it should be infinity since all bobs should match
   return matchBlockWithBlobs(config, allBlocks, allBlobSidecars, Infinity, BlockSource.byRoot);
 }
+
+export async function unavailableBeaconBlobsByRoot(
+  config: ChainForkConfig,
+  network: INetwork,
+  peerId: PeerIdStr,
+  unavailableBlockInput: BlockInput
+): Promise<BlockInput> {
+  if (unavailableBlockInput.type !== BlockInputType.blobsPromise) {
+    return unavailableBlockInput;
+  }
+
+  const blobIdentifiers: deneb.BlobIdentifier[] = [];
+  const {block, blobsCache, resolveAvailability, blockBytes} = unavailableBlockInput;
+
+  const slot = block.message.slot;
+  const blockRoot = config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block.message);
+
+  const blobKzgCommitmentsLen = (block.message.body as deneb.BeaconBlockBody).blobKzgCommitments.length;
+  for (let index = 0; index < blobKzgCommitmentsLen; index++) {
+    if (blobsCache.has(index) === false) blobIdentifiers.push({blockRoot, index});
+  }
+
+  let allBlobSidecars: deneb.BlobSidecar[];
+  if (blobIdentifiers.length > 0) {
+    allBlobSidecars = await network.sendBlobSidecarsByRoot(peerId, blobIdentifiers);
+  } else {
+    allBlobSidecars = [];
+  }
+
+  // add them in cache, this way may be wait on availability might be resolved
```

Comment:
"may be wait on availability might be resolved" what does this refer to?

Reply:
Ah right so we need to update the cache to resolve the availability promise here
https://github.com/ChainSafe/lodestar/blob/484b9fd78d0e6ba66ffdbc1e8c2b92ccaf3864b4/packages/beacon-node/src/chain/blocks/verifyBlocksDataAvailability.ts#L120

which might already resolve in the same function call here
https://github.com/ChainSafe/lodestar/blob/fa9f16d0d33dae717f45bb1f6dbec7bc8a2893b2/packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRoot.ts#L86

or the other case is when we receive blocks a block / blobs from gossip here
https://github.com/ChainSafe/lodestar/blob/fa9f16d0d33dae717f45bb1f6dbec7bc8a2893b2/packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts#L115-L116

and updating the cache means we might resolve it in either case